src/handlers/wallet/get_exchange_url.rs

```diff
@@ -0,0 +1,140 @@
+use {
+    crate::handlers::wallet::exchanges::{ExchangeError, ExchangeType, GetBuyUrlParams},
+    crate::{
+        handlers::{SdkInfoParams, HANDLER_TASK_METRICS},
+        state::AppState,
+        utils::crypto::{disassemble_caip10, Caip19Asset},
+    },
+    axum::{
+        extract::{ConnectInfo, Query, State},
+        Json,
+    },
+    hyper::HeaderMap,
+    serde::{Deserialize, Serialize},
+    std::{net::SocketAddr, sync::Arc},
+    thiserror::Error,
+    tracing::debug,
+    wc::future::FutureExt,
+};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GeneratePayUrlRequest {
+    pub exchange_id: String,
+    pub asset: String,
+    pub amount: String,
```

Comment:
I'd use `U256` for for easy hex parsing numbers and bounds checking. Also `Bytes` when no bounds required.

```suggestion
    pub amount: alloy::primitives::U256,
```